tests/test_ldd.py

```diff
@@ -0,0 +1,244 @@
+        # yapf: disable
+        lines = [
+            "libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f9a19d8a000)",
+            "/lib64/ld-linux-x86-64.so.2 (0x00007f9a1a329000)",
+            "libboost_program_options.so.1.62.0 => not found",
+            "linux-vdso.so.1 =>  (0x00007ffd7c7fd000)",
+            "libopencv_stitching.so.3.3 => not found",
+            "libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f4b78462000)",
+            "libz.so.1 => not found",
+            "../build/debug/libextstr.so => not found",
+            "/home/user/lib/liblmdb.so => not found"
+        ]
+        # yapf: enable
+        expected_deps = [
+            lddwrap.Dependency(
+                soname="libstdc++.so.6",
+                path=pathlib.Path("/usr/lib/x86_64-linux-gnu/libstdc++.so.6"),
+                found=True,
+                mem_address=""),
+            lddwrap.Dependency(
+                soname=None,
+                path=pathlib.Path("/lib64/ld-linux-x86-64.so.2"),
+                found=True,
+                mem_address=""),
+            lddwrap.Dependency(
+                soname="libboost_program_options.so.1.62.0",
+                path=None,
+                found=False,
+                mem_address=None),
+            lddwrap.Dependency(
+                soname="linux-vdso.so.1", path=None, found=True,
+                mem_address=""),
+            lddwrap.Dependency(
+                soname="libopencv_stitching.so.3.3",
+                path=None,
+                found=False,
+                mem_address=None),
+            lddwrap.Dependency(
+                soname="libstdc++.so.6",
+                path=pathlib.Path("/usr/lib/x86_64-linux-gnu/libstdc++.so.6"),
+                found=True,
+                mem_address=""),
+            lddwrap.Dependency(
+                soname="libz.so.1", path=None, found=False, mem_address=None),
+            lddwrap.Dependency(
+                soname=None,
+                path=pathlib.Path("../build/debug/libextstr.so"),
+                found=False,
+                mem_address=None),
+            lddwrap.Dependency(
+                soname=None,
+                path=pathlib.Path("/home/user/lib/liblmdb.so"),
+                found=False,
+                mem_address=None)
+        ]
+
+        for i, line in enumerate(lines):
+            dep = lddwrap._parse_line(line=line)  # pylint: disable=protected-access
+
+            self.assertTrue(
+                dependencies_equal(dep, expected_deps[i]),
+                "Incorrect dependency read from the line {}, expected {}\ngot {}"
+                .format(line, expected_deps[i], dep))
+
+    def test_parse_wrong_line(self):
+        # ``parse_line`` raises a RuntimeError when it receives an unexpected structured line
+
+        line = "some wrong data which does not make sense"
+
+        self.assertRaises(RuntimeError, lddwrap._parse_line, line)  # pylint: disable=protected-access
```

Comment:
Pleas check for the error message. This is necessary since your current test does not check what the cause of the error is -- it could have been that RuntimeError was raised for a different reason.

Please ask Adam to show you our `testerr` live template.